src/auth/multi-factor.ts

```diff
@@ -0,0 +1,144 @@
+/*!
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as utils from '../utils';
+import { AuthClientErrorCode, FirebaseAuthError } from '../utils/error';
+import { MultiFactorId, MultiFactorInfoResponse } from './multi-factor-internal';
+
+/**
+ * Interface representing the common properties of a user enrolled second factor.
+ */
+export abstract class MultiFactorInfo {
+  /**
+   * The ID of the enrolled second factor. This ID is unique to the user.
+   */
+  public readonly uid: string;
+
+  /**
+   * The optional display name of the enrolled second factor.
+   */
+  public readonly displayName?: string;
+
+  /**
+   * The type identifier of the second factor. For SMS second factors, this is `phone`.
+   */
+  public readonly factorId: MultiFactorId;
```

Comment:
Shouldn't `MultiFactorId` type be exposed?

Reply:
@bojeil-google In `auth.d.ts` we have `factorId` as `string` while here we expose an enum. Do you have any thoughts on the best way forward?

Reply:
Either way is fine. Currently, this can only have the value `phone`. But in the future we will add new types.
I think ideally an enum is preferred.